docs/spirv-fuzz.md

```diff
@@ -0,0 +1,83 @@
+# Guide to writing a spirv-fuzz fuzzer pass
+
+Writing a spirv-fuzz fuzzer pass usually requires two main contributions:
+
+- A *transformation*, capturing a small semantics-preserving change that can be made to a SPIR-V module.  This requires adding a protobuf message representing the transformation, and a corresponding class that implements the `Transformation` interface.
+- A new *fuzzer pass* class, implementing the `FuzzerPass` interface, that knows how to walk a SPIR-V module and apply the new transformation in a randomized fashion.
+
+In some cases, more than one kind of transformation is required for a single fuzzer pass, and in some cases the transformations that a new fuzzer pass requires have already been introduced by existing passes.  But the most common case is to introduce a transformation and fuzzer pass together.
```

Comment:
Remove: "and in some cases the transformations that a new fuzzer pass requires have already been introduced by existing passes."

A fuzzer pass does not really require transformations; it requires stuff in the SPIR-V module. So I think this is just confusing. Perhaps it would be better to replace it with a mini-example.

"In some cases, more than one kind of transformation is required for a single fuzzer pass; e.g. transformation that adds a type declaration, transformation that adds a use of the type declaration."